test/support/test_utils.ex

```diff
@@ -0,0 +1,11 @@
+defmodule Klife.TestUtils do
+  use Retry
+
+  def wait_for_broker_connection(cluster_name) do
+    wait constant_backoff(50) |> expiry(1_000) do
+      :persistent_term.get({:known_brokers_ids, cluster_name}, false)
+    end
+
+    :ok
+  end
```

Comment:
I think we do not need to add the `Retry` dependency only for this simple use case.

We can do something like this:

```elixir
def wait_for_broker_connection(cluster_name, timeout \\ :timer.seconds(5)) do
  deadline = System.monotonic_time() + System.convert_time_unit(timeout, :millisecond, :native)
  do_wait_for_broker_connection(cluster_name, deadline)
end

defp do_wait_for_broker_connection(cluster_name, deadline) do
  if System.monotonic_time() <= deadline do
    case :persistent_term.get({:known_brokers_ids, cluster_name}, :not_found) do
      :not_found ->
        Process.sleep(50)
        do_wait_for_broker(cluster_name, deadline)
      data ->
        data
    end
  else
    raise "timeout while waiting for broker connection"
  end
end
```

This is good enough for tests purposes IMO.